src/sisp.test.js

```diff
@@ -59,12 +59,218 @@ describe('SISP Payment', () => {
+          merchantRespMerchantSession: 'S20200626121231',
+          merchantRespMessageID: 'Zo23nJ2P9C3i3e5Z70K1',
+          merchantRespPan: '************4242',
+          merchantRespPurchaseAmount: '400.00',
+          merchantRespReferenceNumber: '',
+          merchantRespReloadCode: '',
+          merchantRespTid: '00004787',
+          merchantRespTimeStamp: '2020-06-26 12:11:01',
+          messageType: 'P',
+          resultFingerPrint: '623aUzWxGVU8Stp8hlck9hLZo9/+T9xvs3lxNDkDwa2z2PQWF25LO9IjozPtdU95dlG8eAMTGj2cW+2QpIQ+4w==',
+          resultFingerPrintVersion: '1'
+        };
+
+        const referenceId = '1:3:testForm';
+        const total = 89;
+
+        const sisp = new Sisp({ posID, posAutCode });
+        const requestFormResult = await sisp.generatePaymentRequestForm(referenceId, total, webhookUrl);
+
+        const formExpected = successData.replace(/\s/g, '');
+
+        const result = await sisp.paymentProcessedSuccessfully(paymentSuccessData);
+        const expected = { code: 200, message: 'Payment processing success', data: paymentSuccessData };
+
+        expect(requestFormResult.replace(/\s/g, '')).toEqual(formExpected);
+        expect(result).toEqual(expected);
+      });
+    });
+    describe('Error Payment', () => {
+      it('should fail if messageType is different from "8", "10", "M" or "P"', async () => {
+        const paymentSuccessData = {
+          languageMessages: 'pt',
+          merchantResp: 'C',
+          merchantRespAdditionalErrorMessage: '',
+          merchantRespCP: '1047',
+          merchantRespClientReceipt: '',
+          merchantRespEntityCode: '',
+          merchantRespErrorCode: '',
+          merchantRespErrorDescription: '',
+          merchantRespErrorDetail: '',
+          merchantRespMerchantRef: '52',
+          merchantRespMerchantSession: 'S20200626121231',
+          merchantRespMessageID: 'Zo23nJ2P9C3i3e5Z70K1',
+          merchantRespPan: '************4242',
+          merchantRespPurchaseAmount: '400.00',
+          merchantRespReferenceNumber: '',
+          merchantRespReloadCode: '',
+          merchantRespTid: '00004787',
+          merchantRespTimeStamp: '2020-06-26 12:11:01',
+          messageType: 'A',
+          resultFingerPrint: 'GXtUrKOwPtZwTbGiuDEjERoPvsmAUOCB9HRC2Smr+/VocZVluiTzmgbquPJpjPz3yuW0Ouhz05lRWUfF02mJtA==',
+          resultFingerPrintVersion: '1'
+        };
+
+        const referenceId = '1:3:testForm';
+        const total = 89;
+
+        const sisp = new Sisp({ posID, posAutCode });
+        const requestFormResult = await sisp.generatePaymentRequestForm(referenceId, total, webhookUrl);
+
+        const formExpected = successData.replace(/\s/g, '');
+
+        const result = sisp.paymentProcessedSuccessfully(paymentSuccessData);
+        const expected = { code: 400, message: 'Payment processing error' };
+
+        expect(requestFormResult.replace(/\s/g, '')).toEqual(formExpected);
+        expect(result).toEqual(expected);
```

Comment:
Maybe is better we add something like this to be more aligned with the description: expect(result.data.messageType).toEqual(expected.data.messageType) for each success message; then if this fail we will se the message for example: this expected 8 but got 50. What do you think @FredSoares

Reply:
@savylopes If this fail `Jest` will show the expected response and the received response with their differences

Reply:
Yes. You right